src/neuroglancer/ui/default_viewer_setup.ts

```diff
@@ -20,16 +20,84 @@ import {setDefaultInputEventBindings} from 'neuroglancer/ui/default_input_event_
 import {makeDefaultViewer} from 'neuroglancer/ui/default_viewer';
 import {bindTitle} from 'neuroglancer/ui/title';
 import {UrlHashBinding} from 'neuroglancer/ui/url_hash_binding';
+import {UserLayer, UserLayerConstructor} from 'neuroglancer/layer';
+import {SegmentationUserLayer} from 'neuroglancer/segmentation_user_layer';
+import {Tool, restoreTool} from 'neuroglancer/ui/tool';
 
 declare var NEUROGLANCER_DEFAULT_STATE_FRAGMENT: string|undefined;
 
+type CustomBinding = {
+  layer: string, tool: string, protocol?: string,
```

Comment:
`tool` should be a json object, not just a string, as some tools support options